rest/app.py

```diff
@@ -71,16 +74,27 @@ def get_task(task_id):
 
 @app.route('/tasks', methods=['POST'])
 def add_task():
-    tasks.append(TASK_TO_ADD)
-    return json.dumps(tasks)
+    task = json.loads(request.data)
+    tasks.append(task)
+    return render_template('index.html'), 201
 
 
 @app.route('/tasks/<int:task_id>', methods=['DELETE'])
 def delete_task(task_id):
     for task in tasks:
         if task['id'] == task_id:
             tasks.remove(task)
-            return json.dumps(users)
+            return json.dumps(tasks)
+    return not_found(404)
+
+
+@app.route('/tasks/<int:task_id>', methods=['PUT'])
+def update_task(task_id):
+    data = dict(request.form.items())

```

Comment:
It's the answer why we have to use request.form
http://docs.python-requests.org/en/master/user/quickstart/#more-complicated-post-requests

Please, Check _json_ instead of _data_ or _params_ for request from rest_client (requests.post(url, json={'id': 1, ...}))
Maybe you'll be able to use _request.json_ instead of _request.form_ for receiving